sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -70,6 +73,19 @@ class CollationStringExpressionsSuite extends QueryTest with SharedSparkSession
     })
   }
 
+  test("REPEAT check output type on explicitly collated string") {
+    def testRepeat(expected: String, stringType: Integer, input: String, n: Integer): Unit = {
+      val s = Literal.create(input, StringType(stringType))
+
+      checkEvaluation(Collation(StringRepeat(s, Literal.create(n))).replacement, expected)
```

Comment:
Does `checkEvaluation` check the output type?

Reply:
Yes, I test if the output type matches the expected result. You can see below that I test if the result has type for example `expected="UNICODE_CI`